pkg/policy/api/cidr.go

```diff
@@ -75,19 +68,31 @@ func (s CIDRSlice) GetAsEndpointSelectors() EndpointSelectorSlice {
 	// If multiple CIDRs representing reserved:world are in this CIDRSlice,
 	// we only have to add the EndpointSelector representing reserved:world
 	// once.
-	var hasWorldBeenAdded bool
+	var hasIPv4AllBeenAdded, hasIPv6AllBeenAdded bool
 	slice := EndpointSelectorSlice{}
 	for _, cidr := range s {
-		if cidr.MatchesAll() && !hasWorldBeenAdded {
-			hasWorldBeenAdded = true
-			slice = append(slice, ReservedEndpointSelectors[labels.IDNameWorld])
+		if cidr == ipv4All {
+			hasIPv4AllBeenAdded = true
+		}
+		if cidr == ipv6All {
+			hasIPv6AllBeenAdded = true
 		}
 		lbl, err := cidrpkg.IPStringToLabel(string(cidr))
 		if err == nil {
 			slice = append(slice, NewESFromLabels(lbl))
 		}
 		// TODO: Log the error?
 	}
+	// We want to add the world selector whenever possible as it limits the growth of the
+	// ipcache.
```

Comment:
Now I'm second-guessing whether there's really an optimization to make here or not. The thinking I had when I raised the question was whether the "allow to world" could become an "allow all" match, and hence be encoded with one entry in the datapath. But in reality we can only inject that match if the policy says "toEntities: all", ie matching also identities within the cluster. Maybe the best that this will achieve is just to shortcut some of the selection logic for the case where it's single stack and the broadest CIDR is allowed by a policy.

I'm not sure this will actually impact either ipcache or policymaps size at this point.

Reply:
(The logic is probably fine either way, the latest version here is a little more complicated but still correct. It's just the comment that I think may not be accurate).

Reply:
Ah, yes. That's not going to impact the policy map size because we *append* the world entity to the CIDR slice.

But then, why do we even do that in the first place? Doesn't the 0.0.0.0/0 CIDR identity already match all traffic?

(I'm a bit worried that, if we don't do that to optimize, then maybe it's nothing we can't just skip doing. And maybe it explains the previous CI failure in ConformanceDatapath.)

Reply:
yeah I'd expect `0.0.0.0/0` CIDR identity to match all (IPv4) traffic already. Maybe we didn't need the extra world selector insertion after all, we just never revisited this. I think it's important that the _identities_ we create have the `reserved:world` label to make sure that any `toEntities: world` policy matches those identities, but I don't think we need to inject the `reserved:world` labelSelector into the policy parsing piece.